### PR TITLE
fix(@angular/cli): support default options for multiselect list x-prompt

### PR DESCRIPTION
**Pull request by pawelfras**
## PR Checklist

Please check to confirm your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular-cli/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #28843

## What is the new behavior?

This PR provides support for default options for multiselect list x-prompt after switching to `@inquirer/prompts` library.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/angular/cli/src/command-builder/schematics-command-module.ts [hunk 1]
```diff
@@ -204,14 +204,15 @@ export abstract class SchematicsCommandModule
...
-                default: definition.default,
+                default: definition.multiselect ? undefined : definition.default,
                 choices: definition.items?.map((item) =>
                   typeof item == 'string'
                     ? {
                         name: item,
                         value: item,
                       }
                     : {
+                        ...item,
```
Comment (pawelfras) [inline packages/angular/cli/src/command-builder/schematics-command-module.ts:215]: Include options that are in line with the new API: https://github.com/SBoudrias/Inquirer.js/tree/main/packages/checkbox#choice-object

File: packages/angular/cli/src/command-builder/schematics-command-module.ts [hunk 2]
```diff
@@ -204,14 +204,15 @@ export abstract class SchematicsCommandModule
 
                   return definition.validator(Object.values(values).map(({ value }) => value));
                 },
-                default: definition.default,
+                default: definition.multiselect ? undefined : definition.default,
```
Comment (pawelfras) [inline packages/angular/cli/src/command-builder/schematics-command-module.ts:207]: `default` property [is still valid](https://github.com/SBoudrias/Inquirer.js/blob/main/packages/select/README.md#options) for `select` package, used for prompt with type `list` which does not support multiselect: https://github.com/pawelfras/angular-cli/blob/62d5c330c0457a3d753359b8f520bc912ed2c3e2/packages/angular/cli/src/command-builder/schematics-command-module.ts#L197

**Review by clydin — APPROVED** [review]
Thank you for the contribution

**Comment by jkrems** [comment]
The changes were merged into the following branches: main, 18.2.x, 19.0.x

**Pull request merged**